pkg/controller/azdrivernode.go

```diff
@@ -70,6 +75,7 @@ func (r *ReconcileAzDriverNode) Reconcile(ctx context.Context, request reconcile
 		if err != nil && !errors.IsNotFound(err) {
 			return reconcile.Result{Requeue: true}, err
 		}
+		r.SharedState.nodeDiskAvailabilityMap.Delete(request.Name)
```

Comment:
For symetry, consider making a `deleteNodeFromDiskAvailabilityMap`

Reply:
Done.